hadoop-ozone/s3gateway/src/test/java/org/apache/hadoop/ozone/s3/endpoint/TestObjectPut.java

```diff
@@ -288,4 +297,80 @@ public void testEmptyStorageType() throws IOException, OS3Exception {
     //default type is set
     Assert.assertEquals(ReplicationType.RATIS, key.getReplicationType());
   }
+
+  @Test
+  public void testDirectoryCreation() throws IOException,
```

Comment:
These tests have significantly more mocking than others in this class. Can we make them more similar to the other tests with less mocking using the `clientStub` and `objectEndpoint` fields that already exist?

Reply:
An idea was to create unit tests for implemented functionality. Would putting those in a different class be more suitable?

Reply:
I think the tests can still be done in in this class. I was just wondering if we could use the existing `clientStub` and `objectEndpoint` instances in this test like the other tests in the class do. If it's too complicated or not as good, for example because the client stub classes do not save directory or FSO information like we need here, the extra mocking could make sense, I was just wondering if there was a reason for doing it this way.

If we do keep it this way we should verify invocation count on the mocked `OzoneBucket#createDirectory` in each test in addition to the return code that is already being checked.

Reply:
I've tried using `clientStub` but it is painful to create bucket with FSO layout. The reason is that there are only methods for creating buckets with default layout. Because of that I prefer not to tinker with those classes but instead use mocks.

I would verify invocation count for `OzoneBucket#createDirectory`.